### PR TITLE
upcoming: [M3-7302] - Replace NodeBalancer detail charts with Recharts

### PR DESCRIPTION
**Pull request by hana-linode**
## Description 📝
This PR is part of a bigger effort to replace chart.js with the more modern Recharts. 

## Changes  🔄
List any change relevant to the reviewer.
- Replace the NodeBalancer details page charts with Recharts
- Refactor code to use one AreaChart component

## Preview 📷
| Before  | After   |
| ------- | ------- |
| <video src="https://github.com/linode/manager/assets/115299789/29ff3568-bb89-4bff-aa5f-55d027355ae0" /> | <video src="https://github.com/linode/manager/assets/115299789/58d1e6a2-07d3-4cf6-8b9b-c3c6cc209c62" /> |
| <video src="https://github.com/linode/manager/assets/115299789/13afd4c3-389b-4bba-a4cf-908fbc246cf8" /> | <video src="https://github.com/linode/manager/assets/115299789/c32de831-c5fc-445e-957f-294840aa0052" /> |

### Prerequisites
(How to setup test environment)
- Have a NodeBalancer that's been running for a while

### Verification steps 
(How to verify changes)
- Go to the Network tab of a NodeBalancer that's been running for a while
  - Confirm the updated UI and that there are no regressions in the graph, units, etc
- Confirm there are no regressions in the Linode Network Transfer History chart from the refactoring

## As an Author I have considered 🤔

*Check all that apply*

- [x] 👀 Doing a self review
- [x] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [x] 🤏 Splitting feature into small PRs
- [x] ➕ Adding a changeset
- [ ] 🧪 Providing/Improving test coverage
- [x] 🔐 Removing all sensitive information from the code and PR description
- [x] 🚩 Using a feature flag to protect the release
- [ ] 👣 Providing comprehensive reproduction steps
- [ ] 📑 Providing or updating our documentation
- [ ] 🕛 Scheduling a pair reviewing session
- [ ] 📱 Providing mobile support
- [ ] ♿  Providing accessibility support

### CODE REVIEW

File: packages/manager/src/components/AreaChart.tsx [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+  }: TooltipProps<any, any>) => {
+    if (active && payload && payload.length) {
+      return (
+        <StyledPaper>
+          <Typography>{tooltipLabelFormatter(label)}</Typography>
+          {payload.map((item) => (
+            <Typography fontFamily={theme.font.bold} key={item.dataKey}>
+              {item.dataKey}: {tooltipValueFormatter(item.value)}
+            </Typography>
+          ))}
```
Comment (hana-linode) [inline packages/manager/src/components/AreaChart.tsx:71]: Handle displaying multiple values (e.g. traffic in and traffic out values for the nodebalancer traffic chart)

image

File: packages/manager/src/features/NodeBalancers/NodeBalancerDetail/NodeBalancerSummary/TablesPanel.tsx [hunk 2]
```diff
@@ -117,6 +156,18 @@ export const TablesPanel = () => {
...
+    const timeData = [];
+
+    // @TODO recharts: remove conditional code and delete old chart when we decide recharts is stable
+    if (flags.recharts && trafficIn) {
+      for (let i = 0; i < trafficIn.length; i++) {
+        timeData.push({
+          'Traffic In': trafficIn[i][1],
+          'Traffic Out': trafficOut[i][1],
+          t: trafficIn[i][0],
+        });
```
Comment (hana-linode) [inline packages/manager/src/features/NodeBalancers/NodeBalancerDetail/NodeBalancerSummary/TablesPanel.tsx:168]: Format data to easily display stacked areas https://recharts.org/en-US/examples/StackedAreaChart

**Review by bnussman-akamai — COMMENTED** [review]
Looks great considering the existing NodeBalancer graphs didn't event work half of the time!


Something seems wrong with the max value on the traffic graph. 
Screenshot 2023-12-12 at 10 37 55 AM

Screenshot 2023-12-12 at 10 35 25 AM

**Review by cpathipa — COMMENTED** [review]
Observation: Looks like like there is a difference in   y-axis / x-axis points compared to prod and this branch. 
| Prod | This Branch|
| ------ | ------- |
|image|image|

Observation: Firewall section is not appearing in this branch.
Not sure if this is expected as part of this change. 
| Prod  | This Branch   |
| ------- | ------- |
|image|image|

**Comment by hana-linode** [comment]
@cpathipa The axis points displayed are expected to be slightly different but the data at each point are still the same

**Review by bnussman-akamai — APPROVED** [review]
Massive improvement considering Chart.js was very broken here!

https://github.com/linode/manager/assets/115251059/b3c5a2a9-af2b-4f31-9b12-ce689793a278

**Comment by carrillo-erik** [comment]
The UI and data looks good. I am noticing a delay when viewing the charts in the NodeBalancer Summary page. I do not experience any delays when looking at the charts in Linodes Transfer tab. However, the same behavior occurs in PROD, will require more investigation for the cause of this.

**Review by carrillo-erik — APPROVED** [review]

**Pull request merged**